include/LeMonADE/utility/AttributeTags.h

```diff
@@ -0,0 +1,215 @@
+ *
+ * @details 
+ *
+ * @todo 
+ **/
+
+
+typedef  int32_t Integer;
+
+
+/*****************************************************************************/
+/**
+ * @file AttributeTag.h
+ *
+ * @brief Class for labels on chains
+ *
+ * @details The class stores the chainID it belongs to and the number of Labels
+ * 	    sitting on the monomer. 
+ *
+ * @todo 
+ **/
+class MonomerLabel
+{
+public:
+  /**
+  * @brief standard constructor for a monomer Label 
+  */
+  MonomerLabel( const int32_t ChainID_, 
+	        const int32_t NLabels_, 
+	        const int32_t LabelID_):
+  ChainID(ChainID_),NLabels(NLabels_),LabelID(LabelID_)
+  { if (ChainID_ < 0 || NLabels_ < 0 || LabelID_ < 0  )
+      throw std::runtime_error("ChainID, NLabels and LabelID should not be smaller 0! "); 
+  }
+  /**
+  * @brief default constructor
+  */
+  MonomerLabel():ChainID(0),NLabels(0),LabelID(0){}
+  /**
+  * @brief copy constructor
+  */
+  MonomerLabel(const MonomerLabel &Label )
+  {
+    
+    ChainID = Label.ChainID;
+    NLabels = Label.NLabels;
+    LabelID = Label.LabelID;    
+  }
+
+  /**
+  * @brief destructor
+  */
+  ~MonomerLabel(){};
+  
+  /**
+  * @brief assignment operator
+  */
+  MonomerLabel& operator= ( const MonomerLabel &Label )
+  { 
+    ChainID = Label.ChainID;
+    NLabels = Label.NLabels;
+    LabelID = Label.LabelID;
+  }
+  
+  /**
+  * @brief equality operator
+  */
+  bool operator == (const MonomerLabel& Label )
+  { 
+    if( Label.ChainID != ChainID ) return false; 
+    if( Label.LabelID != LabelID ) return false; 
+    if( Label.NLabels != NLabels ) return false; 
+    return true;
+  };
+  
+    /**
+  * @brief unequality operator
+  */
+  bool operator != (const MonomerLabel& Label )
+  { 
+    return !(*this == Label );
+  };
+
+  
+  //! set the chain ID the Label belongs to 
+  void     setChainID(const uint32_t ChainID_){ChainID=ChainID_ ;}
+  //! get the chain ID the Label belongs to
+  const uint32_t& getChainID(		       ) const { return ChainID  ;}
+  
+  //! set the number of labels sitting on the monomer 
+  void     setNLabels(const uint32_t NLabels_){NLabels=NLabels_ ;}
+  //! get the number of labels sitting on the monomer 
+  const uint32_t& getNLabels(		       ) const { return NLabels  ;}
+  
+  //! set the ID of the label(s) on the monomer
+  void     setLabelID(const uint32_t LabelID_){LabelID=LabelID_ ;}
+  //! get the ID of the label(s) on the monomer
+  const uint32_t& getLabelID(		       ) const { return LabelID  ;}
+  
+  //! set the chain ID, number of labels and label id 
+  void     setAll(const uint32_t ChainID_,const uint32_t NLabels_,const uint32_t LabelID_)
+  {
+    ChainID=ChainID_ ;
+    NLabels=NLabels_ ;
+    LabelID=LabelID_ ;
+  }
+  //!set all information 
+  void     setAll(const std::vector<uint32_t> in_)
+  {
+    ChainID=in_[0] ;
+    NLabels=in_[1] ;
+    LabelID=in_[2] ;
+  }
+  //! get the chain ID, number of labels and label id 
+  const std::vector<uint32_t>& getAll() const { 
```

Comment:
**LOGICAL ERROR** : reference to local variable
should read `const std::vector<uint32_t> getAll() const`